src/client/java/net/neoforged/neoforge/client/event/RegisterKeyMappingsEvent.java

```diff
@@ -15,24 +21,43 @@
 import org.jetbrains.annotations.ApiStatus;
 
 /**
- * Allows users to register custom {@link net.minecraft.client.KeyMapping key mappings}.
+ * Allows users to register custom {@link KeyMapping key mappings} and {@link KeyMapping.Category key mapping categories}.
  *
- * <p>This event is not {@linkplain ICancellableEvent cancellable}, and does not {@linkplain HasResult have a result}.
+ * <p>This event is not {@linkplain ICancellableEvent cancellable}.
  *
  * <p>This event is fired on the mod-specific event bus, only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
  */
 public class RegisterKeyMappingsEvent extends Event implements IModBusEvent {
     private final Options options;
+    private final List<KeyMapping.Category> categories;
+    private final Map<ResourceLocation, KeyMapping.Category> moddedCategories = new HashMap<>();
 
     @ApiStatus.Internal
-    public RegisterKeyMappingsEvent(Options options) {
+    public RegisterKeyMappingsEvent(Options options, List<KeyMapping.Category> categories) {
```

Comment:
Instead of leaking the category list like this into the event, I would have this as a parameter of `sortAndStoreCategories`